#! /bin/sh -e
## 54_ip32-eth0.dpatch by ilya@total-knowledge.com
##
## DP: meth is built-in ethernet card on O2, and it would make sense for it to be
## DP: eth0, even if there is also another network card in PCI slot.

# Jeff Garzik <jgarzik@pobox.com>

# Date:   Sat, 28 Jan 2006 21:47:04 -0500
# To: Martin Michlmayr <tbm@cyrius.com>
# Cc: netdev@vger.kernel.org, ilya@total-knowledge.com

# NAK.  Imagine if we did this for all platforms...  you would never stop
# churning the Makefile, because there is no perfect order.  Its policy.
# Its just to the distribution, not the kernel, to enforce preferences
# like this.  Don't depend on kernel Makefile driver ordering.


if [ $# -lt 1 ]; then
    echo "`basename $0`: script expects -patch|-unpatch as argument" >&2
    exit 1
fi

[ -f debian/patches/00patch-opts ] && . debian/patches/00patch-opts
patch_opts="${patch_opts:--f --no-backup-if-mismatch} ${2:+-d $2}"

case "$1" in
    -patch) patch -p1 ${patch_opts} < $0;;
    -unpatch) patch -R -p1 ${patch_opts} < $0;;
    *)
        echo "`basename $0`: script expects -patch|-unpatch as argument" >&2
        exit 1;;
esac

exit 0

Index: drivers/net/Makefile
===================================================================
RCS file: /home/cvs/linux/drivers/net/Makefile,v
retrieving revision 1.107
diff -u -r1.107 Makefile
--- a/drivers/net/Makefile        15 Nov 2004 11:49:28 -0000      1.107
+++ b/drivers/net/Makefile        12 Jan 2005 19:58:47 -0000
@@ -28,6 +28,8 @@
 obj-$(CONFIG_MYRI_SBUS) += myri_sbus.o
 obj-$(CONFIG_SUNGEM) += sungem.o sungem_phy.o

+obj-$(CONFIG_SGI_O2MACE_ETH) += meth.o
+
 obj-$(CONFIG_MACE) += mace.o
 obj-$(CONFIG_BMAC) += bmac.o

@@ -125,7 +127,6 @@
 obj-$(CONFIG_SUN3LANCE) += sun3lance.o
 obj-$(CONFIG_DEFXX) += defxx.o
 obj-$(CONFIG_SGISEEQ) += sgiseeq.o
-obj-$(CONFIG_SGI_O2MACE_ETH) += meth.o
 obj-$(CONFIG_AT1700) += at1700.o
 obj-$(CONFIG_FMV18X) += fmv18x.o
 obj-$(CONFIG_EL1) += 3c501.o
